pkgs/os-specific/linux/dracut/default.nix

```
@@ -0,0 +1,100 @@
+{ stdenv
+, lib
+, fetchurl
+, makeWrapper
+, pkg-config
+, kmod
+, asciidoc
+, libxslt
+, docbook5
+, docbook_xsl
+, docbook_xsl_ns
+, coreutils
+, utillinux
+, gnused
+, gnugrep
+, squashfsTools
+, cpio
+, binutils
+, gzip
+, bzip2
+, lz4
+, lzop
+, zstd
+, xz
+}:
+
+stdenv.mkDerivation rec {
+  name = "dracut";
+  version = "049";
+  src = fetchurl {
+    url = "https://github.com/dracutdevs/dracut/archive/${version}.tar.gz";
+    sha256 = "07jz3qlby4npgqgmzxjrp9977r8vhy1igyr3hgvxy0nqj5cx0lvw";
+  };
+  nativeBuildInputs = [
+    makeWrapper
+    pkg-config
+    kmod
+    asciidoc
+    libxslt
+    docbook5
+    docbook_xsl
+    docbook_xsl_ns
+  ];
+  preConfigure = ''
+    patchShebangs ./configure
+  '';
+  postPatch = ''
+    substituteInPlace dracut.sh \
+      --replace 'dracutbasedir=/usr/lib/dracut' "dracutbasedir=$out/lib/dracut"
+    substituteInPlace lsinitrd.sh \
+      --replace 'dracutbasedir=/usr/lib/dracut' "dracutbasedir=$out/lib/dracut"
+  '';
+  postFixup = ''
```

> Any alternative to setting the PATH for all the scripts?

I think @CMCDragonkai you can do something like this (not tested):

```suggestion
  # put here all the packages you need - wrap them all
  wrapPATH = lib.makeBinPath [ ... ];
  postFixup = ''
    wrapProgram $out/bin/dracut --set PATH ${wrapPATH}
  '';
```

BUT, TBH I'm a bit conflicted whether **all** of those packages are needed as they are expected to be available in almost every environment. Personally, I'd drop at least:

- `gzip`
- `bzip2`
- `xz`
- `gnugrep`
- `gnused`

BTW I assume `makeBinPath` uses the order of the packages it gets in order to give precedence in `$PATH`. Hence,  If upstream requires explicitly `gnused` instead whatever's in `coreutils`, you should put `gnused` before `coreutils`.

---

I think the derivation should be as pure as possible to ensure that if I package `dracut` in a docker container it should have everything.